components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -4114,6 +4114,50 @@ public String getServiceProviderNameByClientId(String clientId, String clientTyp
         return applicationName;
     }
 
+    /**
+     * @param inboundKey inbound key
+     * @param inboundType inbound type
+     * @param tenantDomain tenant domain
+     * @return application resource id
+     * @throws IdentityApplicationManagementException
+     */
+    public String getApplicationResourceIDByInboundKey(String inboundKey, String inboundType, String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        int tenantID = getTenantId(tenantDomain);
+        String applicationResourceId = null;
+        // Reading application resource id from the database
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            try (PreparedStatement statement =
+                         connection.prepareStatement(ApplicationMgtDBQueries.LOAD_APP_UUID_BY_CLIENT_ID_AND_TYPE)) {
+                statement.setString(1, inboundKey);
+                statement.setString(2, inboundType);
+                statement.setInt(3, tenantID);
+                statement.setInt(4, tenantID);
+
+                try (ResultSet resultSet = statement.executeQuery()) {
+                    if (resultSet.next()) {
+                        applicationResourceId = resultSet.getString(1);
+                    }
+                }
+            }
+        } catch (SQLException e) {
+            throw new IdentityApplicationManagementServerException("Error while retrieving application " +
+                    "resourceId for inboundKey: " + inboundKey + " in inboundType: " + inboundType +
+                    " in tenantDomain: " + tenantDomain, e);
+        }
+        return applicationResourceId;
+    }
+
+    private int getTenantId(String tenantDomain) throws IdentityApplicationManagementException {
+        try {
+            return IdentityTenantUtil.getTenantId(tenantDomain);
+        } catch (IdentityRuntimeException e) {
+            throw new IdentityApplicationManagementException("Error while retrieving tenant id for retrieve " +
```

Comment:
Message can include the tenant domain

Reply:
Also why do we catch the runtime exception

Reply:
util method throw the runtime exception